graph/xml_parser.py

```diff
@@ -204,23 +208,108 @@ def _operator_as_child(self, value, scan):
         out = dict(
             operator=value['operator'],
             negate=value['negate'],
-            result=value['negate'],
+            result=value['result'],
+            comment=value['comment'],
             node=[])
         for child in value['node']:
             if 'operator' in child:
                 out['node'].append(self._operator_as_child(child, scan))
             elif 'extend_definition' in child:
                 out['node'].append(
                     self._find_definition_by_id(
-                        scan, child['extend_definition'], child['negate']))
+                        scan,
+                        child['extend_definition'],
+                        child['negate'],
+                        child['comment']))
             elif 'value_id' in child:
                 out['node'].append(child)
             else:
                 raise ValueError('error - unknown child')
         return out
 
-    def _find_definition_by_id(self, scan, id, negate_status):
+    def _find_definition_by_id(self, scan, id, negate_status, comment):
         for definition in scan['definitions']:
             if definition['id'] == id:
                 definition['node'][0]['negate'] = negate_status
+                definition['node'][0]['comment'] = comment
                 return self._operator_as_child(definition['node'][0], scan)
+
+    def deeper_in_criteria_comments(self, criteria):
+        if criteria.get('operator') is None:
+            comments = dict(
+                operator='AND',
+                comment=criteria.get('comment'),
+                node=[])
+        else:
+            comments = dict(
+                operator=criteria.get('operator'),
+                comment=criteria.get('comment'),
+                node=[])
+        for criterion in criteria:
+            if criterion.get('operator') is not None:
+                comments['node'].append(
+                    self.deeper_in_criteria_comments(criterion))
+            else:
+                if criterion.get('definition_ref') is not None:
+                    comments['node'].append(
+                        dict(
+                            extend_definition=criterion.get('definition_ref'),
+                            comment=criterion.get('comment')))
+                else:
+                    comments['node'].append(
+                        dict(
+                            value_id=criterion.get('test_ref'),
+                            comment=criterion.get('comment')))
+        return comments
+
+    def prepare_definition_comments(self, oval_definitions):
+        ns = {
+            'oval-definitions': 'http://oval.mitre.org/XMLSchema/oval-definitions-5'
```

Comment:
This looks like a constant that should be together with other namespaces, unless there is a reason not to do it.